torrent.go

```diff
@@ -532,11 +532,17 @@ func (t *Torrent) close() (err error) {
 	if c, ok := t.storage.(io.Closer); ok {
 		c.Close()
 	}
+	for _, tracker := range t.trackerAnnouncers {
+		tracker.stop.Set()
+	}
+	t.trackerAnnouncers = nil
 	for _, conn := range t.conns {
 		conn.Close()
 	}
+	t.conns = nil
+	t.peers = nil
+	t.halfOpen = nil
 	t.pieceStateChanges.Close()
-	t.updateWantPeersEvent()
 	return
```

Comment:
Rather than the above changes, trackerScrapers should just test for Torrent.closed in their main loop. Torrent.addPeers should not trigger new network activity if the Torrent is closed. Torrent.updateWantPeersEvent should still occur, but wantPeers is always false if the torrent is closed.